integrations-and-supported-tools/transformers/notebooks/Neptune_Transformers.ipynb

```diff
@@ -0,0 +1,382 @@
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setting up model and data for training"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from datasets import load_dataset, load_metric\n",
+    "from transformers import AutoTokenizer\n",
+    "from transformers import AutoModelForSequenceClassification, TrainingArguments, Trainer\n",
+    "from transformers.integrations import NeptuneCallback"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Loading the data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "task = \"cola\"\n",
+    "model_checkpoint = \"distilbert-base-uncased\"\n",
+    "batch_size = 16\n",
+    "dataset = load_dataset(\"glue\", task)\n",
+    "metric = load_metric('glue', task)\n",
+    "num_labels = 2"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Create Tokenizer for the model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tokenizer = AutoTokenizer.from_pretrained(model_checkpoint, use_fast=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Preprocess the dataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def preprocess_function(examples):\n",
+    "    return tokenizer(examples['sentence'], truncation=True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "encoded_dataset = dataset.map(preprocess_function, batched=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Instantiate the model for finetuning"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model = AutoModelForSequenceClassification.from_pretrained(model_checkpoint, num_labels=num_labels)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Create the training arguments for model finetuning"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model_name = model_checkpoint.split(\"/\")[-1]\n",
+    "\n",
+    "args = TrainingArguments(\n",
+    "    f\"{model_name}-finetuned-{task}\",\n",
+    "    evaluation_strategy = \"epoch\",\n",
+    "    save_strategy = \"epoch\",\n",
+    "    learning_rate=2e-5,\n",
+    "    per_device_train_batch_size=batch_size,\n",
+    "    per_device_eval_batch_size=batch_size,\n",
+    "    num_train_epochs=3,\n",
+    "    weight_decay=0.01,\n",
+    "    load_best_model_at_end=True,\n",
+    "    report_to=\"none\",\n",
```

Comment:
Are we not using `report_to="neptune"`?

Reply:
`report_to` approach already has a small snippet in the docs: https://docs.neptune.ai/integrations-and-supported-tools/model-training/hugging-face

We want to demonstrate manually passing the `run` approach.

NOTE: If we don't do `report_to=None`, then the trainer will have two callbacks. One that we passed and other created by the `report_to="neptune"`. Side note: `report_to` right now defaults to all (which is painful and leads to confusing errors). In upcoming versions of `transformers`, it will default to "none".

Reply:
Added a note to the (new) docs.